src/main/java/it/gov/pagopa/payhub/activities/service/IuvService.java

```diff
@@ -0,0 +1,109 @@
+package it.gov.pagopa.payhub.activities.service;
+
+import it.gov.pagopa.payhub.activities.dao.IuvSequenceNumberDao;
+import it.gov.pagopa.payhub.activities.dto.OrganizationDTO;
+import it.gov.pagopa.payhub.activities.exception.ValueNotValidException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class exposing methods related to IUV handling.
+ */
+@Lazy
+@Service
+@Slf4j
+public class IuvService {
+
+  private static final String PROGRESSIVO_PADDING_ZEROES_SMALL = "%011d";
+  public static final String AUX_DIGIT = "3";
+
+  private final String informationSystemId;
+
+  private final IuvSequenceNumberDao progressiviVersamentoDao;
+
+  public IuvService(@Value("${iuv.informationSystemId:00}") String informationSystemId,
```

Comment:
What is the `informationSystemId`?

Reply:
The "information system id" is the two-digit code that identifies an information system that could generate IUV for the organization / segregationCode. Since this code is used when composing the IUV (in position 3 and 4 of the generated IUV), it ensures that IUV generated by different information systems will not collide.
From the point of view of IUV, MyPay / PU is just one of the information systems generating IUV, and by default has 00 as its informationSystemId.
This means that:
- every IUV generated by MyPay / PU will have 00 as the 3° and 4° digit;
- every IUV generated externally (i.e. NOT by MyPay) will have 3° and 4° digit != '00';
- if there are multiple external information systems generating IUV for this ente, any of them will use its informationSystemId.

Reply:
is it right to configure it as a global variable or it should change based on the broker?

Reply:
Considering that currently on MyPay the value of "information System id" is hardcoded in the code, I would keep the handling of this value as simple as possible for the moment. If in future there is the need (but I doubt it) to handle different values per broker, we may always introduce this customization.